### PR TITLE
feat(std/path): Align globToRegExp() with bash glob expansion

### PR DESCRIPTION
**Pull request by nayeemrmn**
- feat: Support escaping glob characters
- feat: Support more character classes
- feat: Match characters literally on segment parse failure
- fix: Match nothing for empty globs
- fix: Contain any glob syntax to its path segment
- perf: Remove extraneous separators from generated regex
- doc: Add detailed JSDoc
- chore: Remove old copyright headers

---

More details in the JSDoc.

### CODE REVIEW

**Comment by nayeemrmn** [comment]
@ry Please review.

File: std/fs/expand_glob.ts [hunk 1]
```diff
@@ -167,15 +169,13 @@ export async function* expandGlob(
...
- * Examples:
+ * Example:
  *
- *     for (const file of expandGlobSync("**\/*.ts")) {
- *       console.log(file);
- *     }
- */
+ *      for (const file of expandGlobSync("**\/*.ts")) {
+ *        console.log(file);
+ *      } */
```
Comment (ry) [inline std/fs/expand_glob.ts:178]: seems like the `*/` should be on the following line.

File: std/path/glob.ts [hunk 2]
```diff
@@ -1,5 +1,3 @@
-// globToRegExp() is originall ported from globrex@0.1.2.
-// Copyright 2018 Terkel Gjervig Nielsen. All rights reserved. MIT license.
```
Comment (ry) [inline std/path/glob.ts:2]: This is being removed because this is a new implementation?

Comment (nayeemrmn) [reply std/path/glob.ts:2]: Yes, this one necessarily rewrites the whole thing to achieve:
```
- feat: Match characters literally on segment parse failure
- fix: Contain any glob syntax to its path segment
```

File: std/fs/expand_glob.ts (hunk 1 above)
Comment (nayeemrmn) [reply std/fs/expand_glob.ts]: Done.

**Review by bartlomieju — APPROVED** [review]
LGTM, great PR @nayeemrmn

**Pull request merged**